From: Andrea Bolognani <eof@kiyuko.org>
Date: Thu, 26 Mar 2026 01:41:02 +0100
Subject: Drop handling of secret encryption from libvirtd.service

It breaks libvirtd completely if the secret driver is not
installed, which is a bit harsh. Until the issue is addressed,
leave the functionality out entirely. The secret driver will
handle this gracefully by simply continuing to store secrets
unencrypted, as it had been doing before 12.1.0.

Bug: https://lists.libvirt.org/archives/list/devel@lists.libvirt.org/message/743FQZ6LGNYEREWOSTJJXHVE67GYBAXU/

Forwarded: not-needed
---
 src/remote/libvirtd.service.in | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/remote/libvirtd.service.in b/src/remote/libvirtd.service.in
index cc50555..073555e 100644
--- a/src/remote/libvirtd.service.in
+++ b/src/remote/libvirtd.service.in
@@ -12,8 +12,6 @@ After=libvirtd.socket
 After=libvirtd-ro.socket
 After=libvirtd-admin.socket
 Requires=virtlogd.socket
-Requires=virt-secret-init-encryption.service
-After=virt-secret-init-encryption.service
 Wants=virtlockd.socket
 After=virtlogd.socket
 After=virtlockd.socket
@@ -31,8 +29,6 @@ Conflicts=xendomains.service
 Type=notify-reload
 Environment=LIBVIRTD_ARGS="--timeout 120"
 EnvironmentFile=-@initconfdir@/libvirtd
-Environment=SECRETS_ENCRYPTION_KEY=%d/secrets-encryption-key
-LoadCredentialEncrypted=secrets-encryption-key:@localstatedir@/lib/libvirt/secrets/secrets-encryption-key
 ExecStart=@sbindir@/libvirtd $LIBVIRTD_ARGS
 ExecReload=/bin/kill -HUP $MAINPID
 KillMode=process
